src/api/axiosInstanceFactory.js

```diff
@@ -0,0 +1,35 @@
+import axios from 'axios';
+import { HTTP_STATUS, HEADERS } from '../constants/http.constants';
+
+export const createAxiosInstance = ({ getToken, onUnauthorized }) => {
+  const instance = axios.create({
+    baseURL: import.meta.env.VITE_API_URL,
+    withCredentials: true,
+  });
+
+  instance.interceptors.request.use(
+    (config) => {
+      const token = getToken?.();
+      if (token) {
```

Comment:
Do you think that the `?` part is necessary ?
Because we are creating 1 instance of axios (and this is when the app start) 
We can see if there is something wrong when the app starts for the first time